yt/frontends/swift/io.py

```diff
@@ -66,7 +66,7 @@ def _get_smoothing_length(self, sub_file, pdtype=None, pshape=None):
             pcount = f["/Header"].attrs["NumPart_ThisFile"][ind].astype("int64")
             pcount = np.clip(pcount - si, 0, ei - si)
             # we upscale to float64
-            hsml = f[ptype]["SmoothingLength"][si:ei, ...]
+            hsml = f[ptype]["SmoothingLengths"][si:ei, ...]
```

Comment:
Here we'd probably need to try both keys for backward compatibility. It's okay to give priority to the newest one but it would also help to add a comment specifying (at least broadly) which version of SWIFT made the change.